src/mixed.js

```diff
@@ -139,7 +139,12 @@ SchemaType.prototype = {
   },
 
   cast(value, options = {}) {
-    let resolvedSchema = this.resolve(options)
+    let resolvedSchema = this.resolve(options);
+
+    if(this._oneOfTypeSchema){
```

Comment:
I think this will be neater and easier to implement as a new schema type instead of extending `mixed`, Joi uses `alternatives` maybe we can copy them there? 

That way this logic can be implementing on top of the base logic and maintain the underlying functions like the conditional schema resolution

Reply:
👍